app/gui/src/dashboard/layouts/Drive/Categories/CategoriesProvider.tsx

```diff
@@ -41,7 +24,7 @@ export function CategoriesProvider(props: CategoriesProviderProps): React.JSX.El
       'driveCategory',
       () => {
         const readSavedCategory = () => {
-          const id = categoryIdStore.getState().categoryId
+          const id = getDriveLocation().categoryId
```

Comment:
This provider does not publish `categoryId` and instead dev is encouraged to read from dive store.

But if the category is specified in URL only, we may get discrepancy between `getDriveLocation().categoryId` and `caregory` from this context (because nothing is reading URL query to put it to drive store).

I think the URL query should be synchronized at the highest published point.